FrontEnd/src/components/adminPage/detail-view/UserDetail.jsx

```diff
@@ -1,64 +1,141 @@
 import css from './UserDetail.module.css';
-import Checkbox from '../../checkbox/Checkbox';
-import {useState} from 'react';
 import DeleteModal from './DeleteModal';
-
-const FIELDS = [
-    {
-        id: 'f1',
-        label: 'Прізвище',
-        form_name: 'lastname',
-    },
-    {
-        id: 'f2',
-        label: 'Ім\'я',
-        form_name: 'name',
-    },
-    {
-        id: 'f3',
-        label: 'Електронна пошта',
-        form_name: 'email',
-    },
-    {
-        id: 'f4',
-        label: 'Номер телефону',
-        form_name: 'phone_number',
-    }
-];
-
+import axios from 'axios';
 
 function UserDetail() {
     const [deleteModalActive, setDeleteModalActive] = useState(false);
+    const [error, setError] = useState(null);
+    const [users, setUsers] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const userId = usePathUserId();
+    const [updateSuccess, setUpdateSuccess] = useState(false);
+    const url = `${process.env.REACT_APP_BASE_API_URL}/api/admin/users/${userId}/`;
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const response = await axios.get(url);
+                if (response.status !== 200) {
+                    throw new Error(`HTTP error! Status: ${response.status}`);
+                }
+                setUsers(response.data);
+                setLoading(false);
+            } catch (error) {
+                setError(error.message);
+                setLoading(false);
+            }
+        };
+        fetchData();
 
-    const handleDeleteClick = () => {
-        setDeleteModalActive(true);
+    }, [userId, url]);
+
+    const handleSaveChanges = async () => {
+        try {
+            const response = await axios.put(
+                url,
+                {
+                    name: users.name,
+                    surname: users.surname,
+                    email: users.email,
+                    is_active: users.is_active,
+                    is_staff: users.is_staff,
+                    is_superuser: users.is_superuser
+                },
+            );
+            if (response.status !== 200) {
+                throw new Error(`HTTP error! Status: ${response.status}`);
+            }
+            setUpdateSuccess(true);
+        } catch (error) {
+            console.error('Failed to update user:', error);
+        }
+    };
+
+    if (loading) {
+        return <p>Loading...</p>;
+    }
+
+    if (error) {
+        return <p>Error: {error}</p>;
+    }
+
+    const handleDeleteUser = async () => {
+        try {
+            const response = await axios.delete(url);
+            if (response.status !== 204) {
+                throw new Error(`HTTP error! Status: ${response.status}`);
+            }
+            setUsers([]);
+            navigate('/customadmin/users');
+        } catch (error) {
+            console.error('Failed to delete user:', error);
+        }
```

Comment:
swr mutation https://swr.vercel.app/docs/mutation

Reply:
In my opinion, the usage of user states here is optimal, and mutation is not needed since the code does not make duplicate requests to the server.